### PR TITLE
Deployables index, third attempt

### PR DESCRIPTION
**Pull request by hacxman**
:-)

### CODE REVIEW

File: src/app/controllers/deployables_controller.rb [hunk 1]
```diff
@@ -25,12 +25,41 @@ def index
...
+      Deployable.transaction do 
+        Pool.find(params[:pool_id])
+        # only for it's side effect of raising exception in case of 404
+        catalogs = Catalog.find_all_by_pool_id(params[:pool_id])
+        deployables = catalogs.map { |_c| _c.deployables }.reduce([], &:+)
+
+        perm_deployables = Deployable.
+          list_for_user(current_session, current_user, Privilege::VIEW)
+
+        @deployables = deployables.to_set & perm_deployables.to_set
```
Comment (rwsu) [inline src/app/controllers/deployables_controller.rb:43]: Does the additions above need to be refactored? The code builds @deployables, but it is then not used in the response which I find confusing.

**Comment by rwsu** [comment]
Please see code comment. 

Also there are a couple of extra whitespaces in the last patch. And at some point catalogs needs to be updated. The deployables that are listed there need an href.

**Comment by hacxman** [comment]
I don't see why it needs to be refactored and well, I don't quite understand your comment on @deployables.

**Comment by rwsu** [comment]
My mistake. I misread the if/else clauses. 

Conditional ack on whitespace issue.

**Comment by hacxman** [comment]
New PR -> #255

**Pull request closed without merging**